File: materials/2019-2020/11v/2019-12-18-polymorphism-exercise/rectangle.cc
#include "rectangle.hh"

Rectangle::Rectangle(
    Point const& upper_left,
    Point const& bottom_right)
  : upper_left(upper_left), bottom_right(bottom_right) {}

void Rectangle::draw() const {
  std::cout
    << "Rectangle:\n"
    << "\tUpper left corner: "
    << upper_left
    << '\n'
    << "\tBottom right corner: "
    << bottom_right;
}

double Rectangle::get_width() const {
  return bottom_right.get_x() - upper_left.get_x();
}

double Rectangle::get_height() const {
  return upper_left.get_y() - bottom_right.get_y();
}

double Rectangle::get_perimeter() const {
  return 2 * get_width() + 2 * get_height();
}

double Rectangle::get_area() const {
  return get_width() * get_height();
}

File: materials/2019-2020/11b/2020-02-26-java-testing-exercise/src/main/java/com/oop/Rectangle.java
package main.java.com.oop;

public class Rectangle {
    private final Point p1;
    private final Point p2;

    public Rectangle(Point p1, Point p2) {
        this.p1 = p1;
        this.p2 = p2;
    }

    public double getPerimeter() {
        double s1 = Math.abs(p1.getX() - p2.getX());
        double s2 = Math.abs(p1.getY() - p2.getY());

        return (s1 + s2) * 2;
    }

    public double getArea() {
        double s1 = Math.abs(p1.getX() - p2.getX());
        double s2 = Math.abs(p1.getY() - p2.getY());

        return s1*s2;
    }

    public void printShape() {
        double s1 = Math.abs(p1.getX() - p2.getX());
        double s2 = Math.abs(p1.getY() - p2.getY());

        System.out.println("Triangle");
        System.out.println("AB = CD = " + s1);
        System.out.println("BC = AD = " + s2);
        System.out.println();
        System.out.println("Perimeter = " + this.getPerimeter());
        System.out.println("Area = " + this.getArea());
    }
}
